diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter5/AvoidNestedFunctionsRule.kt

```diff
@@ -65,7 +65,8 @@ class AvoidNestedFunctionsRule(configRules: List<RulesConfig>) : DiktatRule(
     }
 
     private fun isNestedFunction(node: ASTNode): Boolean =
-            node.hasParent(FUN) && node.hasFunParentUntil(CLASS_BODY) && !node.hasChildOfType(MODIFIER_LIST)
+            node.hasParent(FUN) && node.hasFunParentUntil(CLASS_BODY) && !node.hasChildOfType(MODIFIER_LIST) &&
+                    node.getFirstChildWithType(IDENTIFIER) != null
```

Comment:
I suggest to extract `node.getIdentifierName() != null` for `FUN` nodes into a function in `ASTNodeUtils` like `ASTNode.isAnonimousFunction()`. Then here you could call `node.treeParent.isAnonymousFunction()`. It would be much readable.

Reply:
Ok. Added